tests/key_validate_test.py

```diff
@@ -81,6 +83,252 @@ def test_is_public_private_bip32_valid(self):
                 self.assertEqual(is_private_bip32_valid(a, allowable_netcodes=NETWORK_NAMES), None)
                 self.assertEqual(is_public_bip32_valid(a, allowable_netcodes=NETWORK_NAMES), None)
 
+
+    def test_key_limits(self):
+        nc = 'BTC'
+        cc = b'000102030405060708090a0b0c0d0e0f'
+        order = generator_secp256k1.order()
+
+        for k in -1, 0, order, order + 1:
+            with self.assertRaises(ValueError) as cm:
+                Key(secret_exponent=k)
+            err = cm.exception
+            self.assertEqual(err.args[0], 'invalid secret exponent')
```

Comment:
This is a good example of where a custom exception would be useful. You could replace some of the above with the following instead:

```
with self.assertRaises(InvalidSecretExponentException):
    Key(secret_exponent=k)
```

(Rather than checking the value of the exception's message, you can just make sure it raises the correct exception type.)